### DISCUSSION TITLE
From where I start to dig the Laravel cores?

### DISCUSSION THREAD

**Discussion by paulocuambe** [laravel/framework#35022] [Q&A]
Hello there!

I'm trying for a little while to contribute to the Laravel framework but I can't find an issue that I can help. That said, I would like to ask you, regular contributors and maintainers, to help me find my way inside the cores.

I know that I need to start somewhere but I don't know where to start. This would be a great opportunity for me to also level up my PHP knowledge because from what I have seen and used, this codebase is a beauty.

I will be dedicating some of my paid time to read and understand the framework so I can help more people. And as Taylor Otwell says, make the life of fellow developers easier.

Thanks in advance.

**Comment by paras-malhotra** [comment]
@paulocuambe, the framework is divided into services. I think the easiest way to start is to pick up any services that you're already familiar/comfortable with (e.g. database, cache, queues, broadcast, console, etc.). Study the code starting from the service provider and then mapping that back to the concrete implementation.

Another useful thing would be to read through PRs that are already merged/closed and look at what kind of features are and aren't getting merged. Good luck! 👍

**Reply by paulocuambe** [reply]
Thank you. I'll definitely do this.

I won't mark this as an answer not yet to make sure other people can share their opinions.
